src/facade/reply_capture.cc

```diff
@@ -120,6 +120,7 @@ CapturingReplyBuilder::Payload CapturingReplyBuilder::Take() {
 }
 
 void CapturingReplyBuilder::SendDirect(Payload&& val) {
+  has_replied_ = !holds_alternative<monostate>(val);
```

Comment:
did you see fails on tests that made you to add this fix?

Reply:
Not really no. What I was thinking was that when we transfer the payload between two `CapturingReplyBuilder's` through `Apply` it calls this function. But this can be a little problematic. Why? Because `SKIP_LESS` can make the payload a `monostate` depending on the `reply_mode`. So we can't really rely on that and I don't see a clean way to do this, because we use `take` to extract the payload from `CapturingReplyBuilder` but that payload does not encapsulate the (`has_replied_`) info. Ok, maybe we can add an exception for that (introduce another struct `has_replied_monostate` to encapsulate that) but since this is not exercised anyway -- I rather keep it as is and if it ever crashes because of the monostate I will take care of it :)

Reply:
Doesnt it makes more sence also here not to check if we has replied in ReplyGaurd ? So instead of setting true to has_replied_ here and in SKIPP_LESS  you will  change your logic in ReplyGaurd when to check if there was a reply If reply_mode_ is not FULL

Reply:
We can't really do that because the `needed` is different on each member function:

```
    9 #define SKIP_LESS(needed)     \                        
       has_replied_ = true;                                            
       if (reply_mode_ < needed) { \   

   23 void CapturingReplyBuilder::SendError(ErrorReply error) {                                          
   24   SKIP_LESS(ReplyMode::ONLY_ERR);   
   
   vs
   
   30 void CapturingReplyBuilder::SendMGetResponse(absl::Span<const OptResp> arr) {                      
   31   SKIP_LESS(ReplyMode::FULL);  
```

we skip under different conditions (otherwise we capture it). At the construction of ReplyGuard we know the `reply_mode_` but we don't know which member function it's going to be used when we Invoke the command so we can't really set expectations. For example, assume that `reply_mode_=ReplyMode::ONLY_ON_ERROR` and we `InvokeCmd`. If we call `SendError`, we capture it. But if we call `SendNullArray` we don't! Since we don't know upfront which of the two will happen, then we can't really know if the ReplyGuard should `DCHECK` or not. 

The big issue is though is, that the `CapturingReplyBuilder` only captures what we sent. We need to use `Apply` to actually call the underline `ReplyBuilder::Send` and set `has_replied_` to `true` but unfortunately this happens after we call `InvokeCmd` and after the `ReplyGuard` has exited the scope. Basically, we would still set `has_replied_` to true, in the `Capture` function which again is not what you are suggesting.